keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -93,6 +98,7 @@ class ApplicationController(
     path = ["/{application}/environment/{environment}/constraints"],
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
+  @PreAuthorize("@authorizationSupport.hasApplicationPermission('READ', 'APPLICATION', #application)")
```

Comment:
why does here and below there are only 2 quotes `"@authorizationSupport..` as oppose to the rest that have 4 `"""@authorizationSupport..` ?

Reply:
Because this has a single permission check, as opposed to the other ones which have two and I wanted to break in two lines for readability. It's just a string (a SpEL expression, to be precise), in both cases.